src/app/features/create/components/hydrate-new-vehicle-record/hydrate-new-vehicle-record.component.ts

```diff
@@ -25,19 +33,54 @@ export class HydrateNewVehicleRecordComponent {
 
   get vehicle$(): Observable<VehicleTechRecordModel | undefined> {
     return this.technicalRecordService.editableVehicleTechRecord$.pipe(
-      tap(techRecord => {
-        if (!techRecord) this.navigateBack();
+      tap(vehicle => {
+        if (!vehicle) this.navigateBack();
+        this.vehicleType = vehicle?.techRecord[0].vehicleType;
       })
     );
   }
 
-  navigateBack() {
-    this.globalErrorService.clearErrors();
-    this.router.navigate(['..'], { relativeTo: this.route });
+  get customSectionForms(): Array<CustomFormGroup> {
+    if (this.summary && this.vehicleType) {
+      const commonCustomSections = [this.summary.body.form, this.summary.dimensions.form, this.summary.tyres.form, this.summary.weights.form];
+
+      switch (this.vehicleType) {
+        case VehicleTypes.PSV:
+          return [...commonCustomSections, this.summary.psvBrakes!.form];
+        case VehicleTypes.HGV:
+          return commonCustomSections;
+        case VehicleTypes.TRL:
+          return [...commonCustomSections, this.summary.trlBrakes!.form];
+        default:
+          return [];
+      }
+    } else return [];
+  }
+
+  handleFormState(): void {
+    const form = this.summary?.sections.map(section => section.form).concat(this.customSectionForms);
+    if (form) {
+      this.isInvalid = this.isAnyFormInvalid(form);
+    }
   }
 
   handleSubmit() {
-    this.store.dispatch(createVehicleRecord());
-    this.actions$.pipe(ofType(createVehicleRecordSuccess), take(1)).subscribe(() => this.navigateBack());
+    this.handleFormState();
+    if (!this.isInvalid) {
+      this.store.dispatch(createVehicleRecord());
+      this.actions$.pipe(ofType(createVehicleRecordSuccess), take(1)).subscribe(() => this.navigateBack());
+    }
+  }
+
+  isAnyFormInvalid(forms: Array<CustomFormGroup | CustomFormArray>): boolean {
```

Comment:
I'd be interested to know if similar methods exist on the amend vehicle component and whether we could make them generic

Reply:
Can I look into this as a refactor?

Reply:
That involves extracting some logic from this component and the one you mentioned into a generic class which then these will inherit. That is a great suggestion, but not MVP at this moment and is beyond the scope of this ticket.

Should mention it in the sprint retro though! We've had quite a few refactoring PRs lately.